### PR TITLE
hdos: init at 8

### LINKED ISSUE #307154: Package request: HDOS
**Project description**

HDOS is a propritery client for Old School Runescape that has better performance and graphics 

**Metadata**

* homepage URL: https://hdos.dev/
* source URL: N/A
* license: proprietary
* platforms: linux, darwin, possibly BSDs via Java? 

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by Misterio77**
## Description of changes

Package [HDOS](https://hdos.dev/), a (officially-approved) third party client for [Oldschool RuneScape](https://oldschool.runescape.com/).

Fixes #307154

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by nonetrix** [comment]
Really dumb question, but if I wanted to test this for myself without changing my whole nixpkgs input in my `flake.nix` how would I do that? Also, if I did that I am pretty sure I wouldn't get binary cache so I would have to build 2523 packages or so if Neofetch is to be trusted

**Comment by Qyriad** [comment]
@nonetrix the easiest way is to just checkout Nixpkgs and this PR. PRs can be checked out with the [Github CLI](https://cli.github.com/), or with Git itself with:
 ```bash
$ git fetch origin refs/pull/312710/head && git checkout FETCH_HEAD
````
You may or may not get binary cache for its dependencies based on the whims of what Hydra happens to have already build on this PR's parent, but you can also checkout a commit that is definitely cached in Hydra, and cherry-pick this PR:

```bash
$ git checkout nixpkgs-unstable && git fetch origin refs/pull/312710/head && git cherry-pick FETCH_HEAD
```

The downside of cherry-picking is that it tests this package definition out of context — i.e., it tests that this package definition is valid in general, but not that it would build once merged into master.

After any of those, the package can be built with the new CLI with `nix build -f . hdos`, with flakes with `nix build '.#hdos'`, or the old CLI with `nix-build -A hdos`.

File: pkgs/by-name/hd/hdos/package.nix [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+    copyDesktopItems
+  ];
+
+  installPhase = ''
+    runHook preInstall
+    makeWrapper ${openjdk11}/bin/java $out/bin/hdos \
```
Comment (Qyriad) [inline pkgs/by-name/hd/hdos/package.nix:45]:
```suggestion
    makeWrapper ${lib.getExe openjdk11} $out/bin/hdos \
```

**Review by Qyriad — CHANGES_REQUESTED** [review]

**Comment by nonetrix** [comment]
Works perfectly for me thanks, besides the fact it crashes Hyprland... I think that is just Hyprland having a broken XWayland implementation for now due to rewrites lol

**Review by Qyriad — APPROVED** [review]

**Review by thiagokokada — APPROVED** [review]

**Pull request merged**